server.js

```diff
@@ -13,13 +13,31 @@ var port = process.env.PORT || process.env.OPENSHIFT_NODEJS_PORT || 8080,
     mongoURL = process.env.OPENSHIFT_MONGODB_DB_URL || process.env.MONGO_URL,
     mongoURLLabel = "";
 
-if (mongoURL == null && process.env.DATABASE_SERVICE_NAME) {
-  var mongoServiceName = process.env.DATABASE_SERVICE_NAME.toUpperCase(),
-      mongoHost = process.env[mongoServiceName + '_SERVICE_HOST'],
-      mongoPort = process.env[mongoServiceName + '_SERVICE_PORT'],
-      mongoDatabase = process.env[mongoServiceName + '_DATABASE'],
-      mongoPassword = process.env[mongoServiceName + '_PASSWORD']
-      mongoUser = process.env[mongoServiceName + '_USER'];
+if (mongoURL == null) {
+  var mongoHost, mongoPort, mongoDatabase, mongoPassword, mongoUser;
+  // If using plane old env vars via service discovery
+  if (process.env.DATABASE_SERVICE_NAME) {
+    var mongoServiceName = process.env.DATABASE_SERVICE_NAME.toUpperCase();
+    mongoHost = process.env[mongoServiceName + '_SERVICE_HOST'];
+    mongoPort = process.env[mongoServiceName + '_SERVICE_PORT'];
+    mongoDatabase = process.env[mongoServiceName + '_DATABASE'];
+    mongoPassword = process.env[mongoServiceName + '_PASSWORD'];
+    mongoUser = process.env[mongoServiceName + '_USER'];
+
+  // If using env vars from secret from service binding  
+  } else if (process.env.database_name) {
+    mongoDatabase = process.env.database_name;
+    mongoPassword = process.env.password;
+    mongoUser = process.env.username;
```

Comment:
should these be database_password and database_name ?  (i assume the name can be anything we want since it depends on what the user defines the env in the pod as, based on the secret)

Reply:
No, these are the exact name that the template service broker is sending back. It has to be env.database_name and env.password (unless there is some manual mapping involved).

This allows for me to just bind and turn into env vars without getting involved (mapping anything).